### ISSUE TITLE
Docker-compose not working with actual config ?

### ISSUE THREAD

**Issue by lucacalcaterra** [openhab/openhab-cloud#234]
Hi,

doing the tests I was not able to make the services work properly through docker with docker-compose. With actual docker-compose.yaml  (my actual config is ubuntu 18.04 and docker 18.09.1)  mongodb not working because say me missing /data/db (even if i create dir as suggested) and nginx proxy not forward to app-1 because it's volume not shared.

Now with the following config seems to work, if it can help someone... :

````yaml

version: '3'
services:
    app-1:
        container_name: app-1
        image: docker.io/openhab/openhabcloud-app
        working_dir: /opt/openhabcloud
        volumes:
        - app-1:/opt/openhabcloud
        networks:
        - ohcloud-network
        links:
        - mongodb
        - redis
        ports:
        - "3000"
        depends_on:
        - mongodb
        - redis  
        restart: always     
    mongodb:
        container_name: mongodb
        image: bitnami/mongodb:latest
        ports:
        - "27017:27017"
#        command: mongod --smallfiles --bind_ip_all
        networks:
        - ohcloud-network
        volumes:
        - /tmp:/bitnami # change /tmp to data dir what you want
#        - /data/configdb
        restart: always
    redis:
        container_name: redis
        image: bitnami/redis:latest
        networks:
        - ohcloud-network
        ports:
        - "6379"
        environment:
        - REDIS_PASSWORD=123_openHAB
        restart: always
    nginx:
        container_name: nginx
        image: docker.io/openhab/openhabcloud-nginx
        volumes:
        - app-1:/opt/openhabcloud
        networks:
        - ohcloud-network
        ports:
        - "80:8081"
        - "443:8443"
        links:
        - app-1:app-1
volumes:
  app-1:
networks:
  ohcloud-network:
    driver: bridge

**Comment by marziman** [comment]
I will recheck this.
Strange, as I had it running not long ago. But anyhow @lucacalcaterra thanks for the Feedback.

BR Mehmet

**Comment by lucacalcaterra** [comment]
Probably some updated docker's images ?

**Comment by SpaceMaster85** [comment]
I can confirm that the actual docker compose file from the repo is not working on an new docker installation (Ubuntu 18.04). 

@lucacalcaterra Thank you for posting your solution! Your compose file is working flawless!
Maybe the documentation and the docker-compose file can be updated on the repo?

**Comment by lucacalcaterra** [comment]
> I can confirm that the actual docker compose file from the repo is not working on an new docker installation (Ubuntu 18.04).
> 
> @lucacalcaterra Thank you for posting your solution! Your compose file is working flawless!
> Maybe the documentation and the docker-compose file can be updated on the repo?

My docker compose file not working on your Ubuntu 18.04  ? I tested on a 18.04 clean install if i'm not wrong

**Comment by SpaceMaster85** [comment]
Sorry @lucacalcaterra I think you misunderstood me.
Your docker-compose file works without problems!

The original one in the repository from openhab-cloud is not working.

**Comment by lucacalcaterra** [comment]
Ok.. probably i'll post a pull request for integrate It on the repo. Or the
admin of repo can copy It ...

Il Ven 22 Mar 2019, 19:36 SpaceMaster85 <notifications@github.com> ha
scritto:

**Comment by RyanRamchandar** [comment]
Just ran in to this same issue when trying to run the `docker-compose.yml` file from `master`.

It seems that that `mongodb` container switched from UID 0 (root) to UID 1001.
https://github.com/bitnami/bitnami-docker-mongodb/issues/106#issuecomment-412458234

Using @lucacalcaterra's compose file worked for me and I was able to `docker-compose up -d` successfully. However, I ran into #227 while registering and can't continue.

**Comment by alackmann** [comment]
I also found the dockercompose broken and updated with my own. I used Traefik as a proxy instead of nginx though as it provides Let's Encrypt certs out of the box for HTTPS.

Also - is it just me, or has the docker.io/openhab/openhabcloud-app not been rebuilt on docker hub for a year? So, it's fairly out of date now unless you build it yourself right? @marziman or @digitaldan can you setup the webhooks to trigger docker hub rebuilds for new tags and `latest` builds for `master`?

**Comment by RayBa82** [comment]
I also wrote a small docker file to build the latest version.
Its based on the ones in the repo, but the documentation is totally outdated and it doesnt run plug and play as far as I see.

This runs standalone, it checks out the master repository and builds it with npm.

The config file can be mounted with docker-compose, if needed I can also provide and updated compose-files which only opens port 80 for the world and other small improvements.

I think the docker stuff should be moved to another repository.

`FROM mhart/alpine-node:8

RUN apk update && \
	apk upgrade && \
    apk add --no-cache git && \
	apk add --no-cache build-base && \
	apk add --no-cache python

RUN addgroup -S openhabcloud && \
	adduser -H -S -g openhabcloud openhabcloud
    
# Add proper timezone
RUN apk add --no-cache tzdata && \
	cp /usr/share/zoneinfo/Europe/Berlin /etc/localtime && \
	echo "Europe/Berlin" >  /etc/timezone

# Cleanup container
RUN rm -rf \
    /usr/share/man \
    /tmp/* /var/cache/apk/* \
    /root/.npm \
    /root/.node-gyp \
    /root/.gnupg \
    /usr/lib/node_modules/npm/man \
    /usr/lib/node_modules/npm/doc \
    /usr/lib/node_modules/npm/html \
    /usr/lib/node_modules/npm/scripts

WORKDIR /opt
RUN git clone https://github.com/openhab/openhab-cloud.git openhabcloud

WORKDIR /opt/openhabcloud
RUN git checkout master
RUN cp config-development.json config.json
RUN mkdir /data
RUN ln -s /opt/openhabcloud/package.json /data

WORKDIR /data
RUN npm install && npm rebuild bcrypt --build-from-source
ENV NODE_PATH /data/node_modules

RUN chown -R openhabcloud:openhabcloud /opt/openhabcloud
RUN chown -R openhabcloud:openhabcloud /data

USER openhabcloud
WORKDIR /opt/openhabcloud
CMD ["node", "app.js"]
`

**Comment by digitaldan** [comment]
I can try and take a look.  I don't actually use the docker functionality, that was @marziman contribution and i have not looked at it in a long time.

**Comment by alackmann** [comment]
@digitaldan if you modify the docker hub entry to rebuild based on tags, the official image will update anytime the main repo is updated.

Im sure we can then PR some updates to the existing compose file.

**Comment by marziman** [comment]
I will take care for it.
I was just too busy and the images will be automatically build. Will be back on it soon.

BR Mehmet

**Comment by marziman** [comment]
@alackmann,

you are correct the duckerhub images not been updated. But I will introduce a build system, where this will be done automatically.

**Comment by alackmann** [comment]
@marziman I guess you've been too busy. I've forked it for now and setup an automated PR bot to follow the official repo's master. Docker images are here https://cloud.docker.com/repository/docker/alackmann/openhab-cloud

alackmann/openhab-cloud:latest - will follow the official repo
alackmann/openhab-cloud:fork - will have an updated (working) docker-compose script with Traefik/LetsEncrypt. I'll PR these to the main repo also once I've tested more thoroughly

**Comment by marziman** [comment]
Yes indeed,

I was in Saudi Arabia. I am back now and will dedicate some time for it tomorrow. I will take a look on your stuff. As mentioned i will check that the image works properly.

Thanks for being patient!
BR Mehmet

**Comment by alackmann** [comment]
@marziman did you have a chance to look at PR #276 ? Any issues or concerns?

**Comment by alackmann** [comment]
@marziman any chance you might look at that PR this year?

Anyone else, if you need an updated, auto build version of this repo, I've set one up here: https://hub.docker.com/r/alackmann/openhab-cloud

Use:
* `latest` for the official repo
* `fork` if you want my changes (https/Lets Encrypt, config via volume, fixed logging for Docker instances)

**Comment by anoff** [comment]
@alackmann thanks for your effort in providing this fix. Stunned that after a year this issue is still unresolved even though a solution is available 😓

I can confirm that @alackmann's fork works

**Comment by digitaldan** [comment]
@marziman is this something you can look at?  I am fine with #276 , although i have not tried it.

**Comment by digitaldan** [comment]
I have merged #276 , thanks for the contribution and apologies for not reviewing sooner.

**Comment by alackmann** [comment]
Awesome @digitaldan. Fyi that also fixed #278 so you can close that issue as well.

Now that's in master, it would be ideal if someone with rights updates the DockerHub for OpenHAB Cloud (https://hub.docker.com/r/openhab/openhabcloud-app) to autobuild `master` as the `latest` tag. As-is, this is currently 2 years old and very behind master.

**Comment by marziman** [comment]
I will take a look for that @digitaldan ,
sorry I was a bit busy with other stuff.
I can also update the docker hub images. I think I wil add later today a job to build them automatically.
